### PR TITLE
gcp metric alarm ORC-55

### PR DESCRIPTION
**Pull request by WillBeebe**
part of: https://github.com/massdriver-cloud/massdriver/issues/1636

closes ORC-55

### CODE REVIEW

File: gcp-metric-alarm/variables.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
+variable "alarm_configuration" {
+  type = object({
+    metric_type        = string
+    resource_type      = string
+    threshold          = number
+    duration_s         = optional(number, 60)
```
Comment (WillBeebe) [inline gcp-metric-alarm/variables.tf:6]: TF 1.3.x

Comment (WillBeebe) [reply gcp-metric-alarm/variables.tf:6]: (Which we can now run!)

File: gcp-metric-alarm/bool_alarm.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
+resource "google_monitoring_alert_policy" "bool" {
+  count        = var.alarm_configuration.value_type == "BOOL" ? 1 : 0
+  display_name = local.alarm_unique_id
+  combiner     = "OR"
+  conditions {
+    display_name = "${var.md_metadata.name_prefix} BOOL"
```
Comment (coryodaniel) [inline gcp-metric-alarm/bool_alarm.tf:6]: Why aren't we using the var.display_name here? If a package has two bool alarms, they'd have the same name!

File: gcp-metric-alarm/variables.tf (hunk 1 above)
Comment (mclacore) [reply gcp-metric-alarm/variables.tf:6]: What's the 2nd value within `optional(number, 60)`? Is it a multiplier?

Comment (WillBeebe) [reply gcp-metric-alarm/variables.tf:6]: Good q, [here](https://developer.hashicorp.com/terraform/language/expressions/type-constraints#optional-object-type-attributes) are the docs but basically the second value is the `defautl`.

File: gcp-metric-alarm/bool_alarm.tf (hunk 2 above)
Comment (WillBeebe) [reply gcp-metric-alarm/bool_alarm.tf]: Thanks for that, you're totally right. I updated this to be the display name variable with no extra interpolation so it matches the MD UI. I also added some notes around the other... `display_name` attribute that needs to be project-unique for this resource to help future users and future me.

**Pull request merged**